Spigot-Server-Patches/0248-Send-attack-SoundEffects-only-to-players-who-can-see.patch

```diff
@@ -0,0 +1,101 @@
+From 8a86dd45c4b581440c6112d6affd7a0ad74a67c5 Mon Sep 17 00:00:00 2001
+From: Brokkonaut <hannos17@gmx.de>
+Date: Tue, 31 Oct 2017 03:26:18 +0100
+Subject: [PATCH] Send attack SoundEffects only to players who can see the
+ attacker
+
+
+diff --git a/src/main/java/net/minecraft/server/EntityHuman.java b/src/main/java/net/minecraft/server/EntityHuman.java
+index 00fe92f..b76a1e2 100644
+--- a/src/main/java/net/minecraft/server/EntityHuman.java
++++ b/src/main/java/net/minecraft/server/EntityHuman.java
+@@ -282,6 +282,7 @@ public abstract class EntityHuman extends EntityLiving {
+         this.world.a(this, this.locX, this.locY, this.locZ, soundeffect, this.bK(), f, f1);
+     }
+ 
++    public final SoundCategory getSoundCategory() { return this.bK(); } // Paper - OBFHELPER
+     public SoundCategory bK() {
+         return SoundCategory.PLAYERS;
+     }
+@@ -939,6 +940,15 @@ public abstract class EntityHuman extends EntityLiving {
+         this.j = 0;
+     }
+ 
++    // Paper start - send SoundEffect to everyone who can see this entity
++    private void sendSoundEffect(SoundEffect effect) {
++        this.world.sendSoundEffect(this, this.locX, this.locY, this.locZ, effect, this.getSoundCategory(), 1.0F, 1.0F); // This will not send the effect to the player himself
++        if (this instanceof EntityPlayer) {
++            ((EntityPlayer) this).playerConnection.sendPacket(new PacketPlayOutNamedSoundEffect(effect, this.getSoundCategory(), this.locX, this.locY, this.locZ, 1.0F, 1.0F));
++        }
++    }
++    // Paper end
++
+     public void attack(Entity entity) {
+         if (entity.bd()) {
+             if (!entity.t(this)) {
+@@ -963,7 +973,7 @@ public abstract class EntityHuman extends EntityLiving {
+                     int i = b0 + EnchantmentManager.b((EntityLiving) this);
+ 
+                     if (this.isSprinting() && flag) {
+-                        this.world.a((EntityHuman) null, this.locX, this.locY, this.locZ, SoundEffects.fw, this.bK(), 1.0F, 1.0F);
++                        sendSoundEffect(SoundEffects.fw); // Paper - send while respecting visibility
```

Comment:
Is the signature change here (and other places) really necessary? your method that is handling this just grabs this information from elsewhere, I personally would rather see a change of a parameter and the method that it's calling than for a complete signature change

Reply:
@aikar I think it was this one https://github.com/PaperMC/Paper/blob/master/Spigot-Server-Patches/0245-Expose-client-protocol-version-and-virtual-host.patch#L71 but its a bit different because its a variable and not a method

@electronicboy would this be better: https://github.com/Brokkonaut/Paper/blob/NoAttackSoundsFromVanishedPlayersAlternative/Spigot-Server-Patches/0248-Send-attack-SoundEffects-only-to-players-who-can-see.patch ?

Reply:
that's below it, ideally it should be on the same line, but it will still conflict if it changes, which is really what we're concerned about

and yes, that looks better, any merge conflict will be more clear as to what is going on, less likely to be overlooked during a merge should it be changed